selftests/unit/test_test.py

```diff
@@ -98,20 +98,20 @@ def check(uid, name, variant, exp_logdir):
 
     def test_data_dir(self):
         """
-        Tests that a valid datadir exists following the test filename
+        Tests that a valid data dir exists following the test filename
         """
         max_length_name = os.path.join(self.tmpdir, "a" * 250)
         tst = self._get_fake_filename_test(max_length_name)
         self.assertEqual(os.path.join(self.tmpdir, max_length_name + ".data"),
-                         tst.datadir)
+                         tst.get_data('', 'file', False))
 
     def test_no_data_dir(self):
         """
         Tests that with a filename too long, no datadir is possible
         """
         above_limit_name = os.path.join(self.tmpdir, "a" * 251)
         tst = self._get_fake_filename_test(above_limit_name)
-        self.assertFalse(tst.datadir)
```

Comment:
And here the docstring might by synced to say `Checks "get_data" won't report fs-unfriendly data dir name.`